cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -18,72 +18,194 @@ package options
+	KubeCloudShared   *cmoptions.KubeCloudSharedOptions
+	ServiceController *cmoptions.ServiceControllerOptions
+
+	SecureServing *apiserveroptions.SecureServingOptions
+	// TODO: remove insecure serving mode
+	InsecureServing *cmoptions.InsecureServingOptions
+	Authentication  *apiserveroptions.DelegatingAuthenticationOptions
+	Authorization   *apiserveroptions.DelegatingAuthorizationOptions
+
+	Master     string
+	Kubeconfig string
 
 	// NodeStatusUpdateFrequency is the frequency at which the controller updates nodes' status
 	NodeStatusUpdateFrequency metav1.Duration
 }
 
 // NewCloudControllerManagerOptions creates a new ExternalCMServer with a default config.
 func NewCloudControllerManagerOptions() *CloudControllerManagerOptions {
-	componentConfig := cmoptions.NewDefaultControllerManagerComponentConfig(ports.InsecureCloudControllerManagerPort)
+	componentConfig := NewDefaultComponentConfig(ports.InsecureCloudControllerManagerPort)
 
 	s := CloudControllerManagerOptions{
-		// The common/default are kept in 'cmd/kube-controller-manager/app/options/util.go'.
-		// Please make common changes there and put anything cloud specific here.
-		Generic:                   cmoptions.NewGenericControllerManagerOptions(componentConfig),
-		NodeStatusUpdateFrequency: metav1.Duration{Duration: 5 * time.Minute},
+		CloudProvider:    &cmoptions.CloudProviderOptions{},
+		Debugging:        &cmoptions.DebuggingOptions{},
+		GenericComponent: cmoptions.NewGenericComponentConfigOptions(componentConfig.GenericComponent),
+		KubeCloudShared:  cmoptions.NewKubeCloudSharedOptions(componentConfig.KubeCloudShared),
+		ServiceController: &cmoptions.ServiceControllerOptions{
+			ConcurrentServiceSyncs: componentConfig.ServiceController.ConcurrentServiceSyncs,
+		},
+		SecureServing: apiserveroptions.NewSecureServingOptions(),
+		InsecureServing: &cmoptions.InsecureServingOptions{
+			BindAddress: net.ParseIP(componentConfig.KubeCloudShared.Address),
+			BindPort:    int(componentConfig.KubeCloudShared.Port),
+			BindNetwork: "tcp",
+		},
+		Authentication:            nil, // TODO: enable with apiserveroptions.NewDelegatingAuthenticationOptions()
+		Authorization:             nil, // TODO: enable with apiserveroptions.NewDelegatingAuthorizationOptions()
+		NodeStatusUpdateFrequency: componentConfig.NodeStatusUpdateFrequency,
 	}
 
-	s.Generic.SecureServing.ServerCert.CertDirectory = "/var/run/kubernetes"
-	s.Generic.SecureServing.ServerCert.PairName = "cloud-controller-manager"
+	s.SecureServing.ServerCert.CertDirectory = "/var/run/kubernetes"
+	s.SecureServing.ServerCert.PairName = "cloud-controller-manager"
+
+	// disable secure serving for now
+	// TODO: enable HTTPS by default
+	s.SecureServing.BindPort = 0
 
 	return &s
 }
 
+// NewDefaultComponentConfig returns cloud-controller manager configuration object.
+func NewDefaultComponentConfig(insecurePort int32) componentconfig.CloudControllerManagerConfiguration {
```

Comment:
Can we please soon move the controller-manager and ccm componentconfig to dedicated packages/API groups?